app/controllers/members_controller.rb

```diff
@@ -1,13 +1,21 @@
+# frozen_string_literal: true
+
 class MembersController < ApplicationController
+  before_action :set_event, only: %i[create destroy]
+
   def create
-    event = Event.find(params[:event_id])
     event.members << current_user
     redirect_back(fallback_location: root_path)
   end
 
   def destroy
-    event = Event.find(params[:event_id])
     event.members.destroy(current_user)
     redirect_back(fallback_location: root_path)
   end
-end
\ No newline at end of file
+
+  private
+
+  def set_event
+    event = Event.find(params[:event_id])
```

Comment:
I think this is not working right now. You need to use instance variable instead of local variable.